x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderDecisionTests.java

```diff
@@ -501,6 +501,7 @@ private static RoutingAllocation createRoutingAllocation(ClusterState state, All
 
     private void withRoutingAllocation(Consumer<RoutingAllocation> block) {
         RoutingAllocation allocation = createRoutingAllocation(state, createAllocationDeciders());
+        allocation.getGatewayAllocations(); // stop tracking straight away
```

Comment:
It is very confusing that getter call also stops tracking something.
Should there be a dedicated call instead? 

Or may be even explicit method to start tracking instead of listing places that do not need to.
It is very easy to miss a new exception as everything would keep working as is, otherwise if we forget to opt in then something would not work.

Reply:
Yeah I don't like this and it's causing many test failures. Better approach incoming.